libraries/AP_HAL_Linux/Storage.cpp

```diff
@@ -11,6 +11,17 @@
 #include <AP_HAL/AP_HAL.h>
 #include <AP_Vehicle/AP_Vehicle_Type.h>
 
+// Necessary for toolchains that does not provide `strndupa`, such as musl.
+#ifndef __GLIBC__
+static inline char* strndupa(const char *old_string, const size_t size)
+{
+    const size_t string_size = strnlen(old_string, size);
+    char *new_string = static_cast<char*>(alloca(string_size + 1));
+    new_string[string_size] = '\0';
+    return static_cast<char*>(memcpy(new_string, old_string, size));
+}
+#endif
```

Comment:
libraries/AP_Common/missing/  would be the proper place and we should make the ifdef generic rather than do it with __GLIBC__.

Reply:
Done.